packages/backend/src/apps/postman-sms/actions/send-sms/index.ts

```diff
@@ -0,0 +1,95 @@
+import type { IRawAction } from '@plumber/types'
+
+import { ZodError } from 'zod'
+
+import { postmanSmsConfig } from '@/config/app-env-vars/postman-sms'
+import StepError, { GenericSolution } from '@/errors/step'
+import { ensureZodObjectKey, firstZodParseError } from '@/helpers/zod-utils'
+
+import getDataOutMetadata from './get-data-out-metadata'
+import { fieldSchema, postmanMessageSchema } from './schema'
+
+const action = {
+  name: 'Send SMS',
+  key: 'sendSms',
+  description: 'Sends an SMS; display name is set to plumber.gov.sg',
+
+  /**
+   * CAVEAT
+   * ---
+   * THIS IS VERY LIKELY TO CHANGE! We currently only allow sending to the only
+   * template in Plumber's SMS campaign, using our API key.
+   *
+   * But in the future, we want to allow users to provide their own API key and
+   * specify their own campaigns and templates. We can't do this right now
+   * because Postman does not provide an API to query templates / template
+   * variables.
+   */
+  arguments: [
+    {
+      label: 'Recipient phone number',
+      description: 'Include country code prefix, e.g. +6581237123',
+      key: ensureZodObjectKey(fieldSchema, 'recipient'),
+      type: 'string' as const,
+      required: true,
+      variables: true,
+    },
+    {
+      label: 'Message',
+      description: 'Max 1000 characters',
+      key: ensureZodObjectKey(fieldSchema, 'message'),
+      type: 'string' as const,
+      required: true,
+      variables: true,
+    },
+  ],
+
+  getDataOutMetadata,
+
+  async run($) {
+    try {
+      const parsedParams = fieldSchema.parse($.step.parameters)
+
+      const response = await $.http.post(
+        '/campaigns/:campaignId/messages',
+        {
+          recipient: parsedParams.recipient,
+          language: 'english',
+          values: {
+            body: parsedParams.message,
+          },
+        },
+        {
+          urlPathParams: {
+            campaignId: postmanSmsConfig.defaultCampaignId,
+          },
+        },
+      )
+      const message = postmanMessageSchema.parse(response.data)
```

Comment:
this is a bit dangerous since the SMS should have successfully sent out at this stage. The only reason this could throw an error is if postman changes their response, so instead of failing the step, we should log it instead. can try using safeParse instead to prevent termination of step

Reply:
For future reference: the thinking is that if postman changes their response, then the variables the users configured in subsequent steps will be invalid; hence we should error out instead of silently failing. This is at the tradeoff of re-sending the SMS if user retries the step

As discussed, its a non-goal to handle downstream API changes if they do not inform us first (since this should not happen). So will switch to logging and letting this step succeed.

Anyway most likely people should not have steps _after_ sending SMSes